zarr/storage.py

```diff
@@ -746,12 +746,19 @@ def __init__(self, path, normalize_keys=False):
     def _normalize_key(self, key):
         return key.lower() if self.normalize_keys else key
 
+    def _fromfile(self, fn):
```

Comment:
It may be worth adding a docstring that mentions some motivation for why we're pulling this functionality into it's own method. Maybe something like:

```python
""" Read data from a file

Parameters
----------
fn: str
    Filepath to open and read from.

Notes
-----
Subclasses should overload this method to specify any custom
file reading logic.
"""
```